src/networkremote/incomingdataparser.cpp

```diff
@@ -370,3 +380,58 @@ Song IncomingDataParser::CreateSongFromProtobuf(
 
   return song;
 }
+
+void IncomingDataParser::AppendFilesToPlaylist(const pb::remote::Message& msg) {
+  if (files_root_folder_.isEmpty()) {  // should never happen...
+    qLog(Warning) << "Remote root dir is not set although receiving "
+                     "APPEND_FILES request...";
+    return;
+  }
+  QDir root_dir(files_root_folder_);
+  if (!root_dir.exists()) {
+    qLog(Warning) << "Remote root dir doesn't exist...";
+    return;
+  }
+
+  const pb::remote::RequestAppendFiles& req_append = msg.request_append_files();
+  QString relative_path = QString::fromStdString(req_append.relative_path());
+  if (relative_path.startsWith("/")) relative_path.remove(0, 1);
+
+  QFileInfo fi_folder(root_dir, relative_path);
+  if (!fi_folder.exists())
+    qLog(Warning) << "Remote relative path " << relative_path
+                  << " doesn't exist...";
+  else if (!fi_folder.isDir())
+    qLog(Warning) << "Remote relative path " << relative_path
+                  << " is not a directory...";
+  else if (root_dir.relativeFilePath(fi_folder.absoluteFilePath())
+               .startsWith("../"))
+    qLog(Warning) << "Remote relative path " << relative_path
+                  << " should not be accessed...";
+  else {
+    QList<QUrl> urls;
+    QDir dir(fi_folder.absoluteFilePath());
+    for (auto it = req_append.files().cbegin(),
```

Comment:
I think you can use `for (const auto& file : req_append.files())`

Reply:
hum true, I gonna give it a try

Reply:
done